types/lodash/index.d.ts

```diff
@@ -12269,7 +12269,7 @@ declare namespace _ {
          * @param value The value to inspect.
          * @return Returns true if value is empty, else false.
          */
-        isEmpty(value?: any): boolean;
+        isEmpty(value?: any): value is null | undefined;
```

Comment:
The compile error that I am seeing refers to `v2` being `undefined`, not `{}` as per current typings, however, in theory you shouldn't get a compilation error at all. As the isEmpty does filter `null` and `undefined`.

```ts
Argument of type '[T, U | undefined]' is not assignable to parameter of type '[T, U]'.
  Type 'U | undefined' is not assignable to type 'U'.
    Type 'undefined' is not assignable to type 'U'.
```